x-pack/test/functional/page_objects/lens_page.ts

```diff
@@ -561,6 +561,16 @@ export function LensPageProvider({ getService, getPageObjects }: FtrProviderCont
       return buttonEl.click();
     },
 
+    async toggleColumnVisibility(label: string) {
+      const id = `lnsColumns-toggle-${label.replace(/ /g, '-')}`;
+      await testSubjects.click('lnsColumnsButton');
+      await testSubjects.existOrFail(id);
+      const isChecked = await testSubjects.isEuiSwitchChecked(id);
+      await testSubjects.setEuiSwitch(id, isChecked ? 'uncheck' : 'check');
+      await testSubjects.click('lnsColumnsButton');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+    },
```

Comment:
It looks like this is toggling from the menu containing all of the columns, but it would be great if we could also interact with the datagrid header columns directly- this would help test the bug I found.